packages/fxa-settings/src/models/integrations/oauth-redirect-integration.ts

```diff
@@ -0,0 +1,185 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import AuthClient from 'fxa-auth-client/lib/client';
+import { createEncryptedBundle } from '../../lib/crypto/scoped-keys';
+import { OAuthRelier } from '../reliers';
+
+/**
+ * Handles logic around generating a redirect for relying part after a sucessfull password reset.
```

Comment:
Typo nits: 
* ~part~ party
* ~sucessfull~ successful
* Also below, "After a ~passoword~ reset"

I am not sure how I feel about making an API call in the integration. I feel like some of this logic can go in the component with more functional helpers instead of calling an integration method since we're trying to avoid some of the content-server abstraction, and later we can pull the authClient API call into a container component. I might be able to look at some of this in FXA-7308 too.

Reply:
Thanks for catching typos. 

I am all for what you are suggesting, I'll switch this to a IOC design, so there's no direct dependency on the auth client. It's worth noting that this function is referenced in two places currently and is a nice abstraction for a PasswordReset operation. As long as we don't end up breaking the abstraction or having to copy and paste between components it sounds good to me.